server-function/src/serveBinaris.ts

```diff
@@ -1,91 +1,82 @@
-  const url = ctx.request.path;
-  const decision = await server.handle(url, ctx.request.headers);
-  switch (decision.action) {
-    case 'handleInvoke': {
-      // TODO: check for allowed origins when supporting CORS
-      // Currently expected to 400 on a Preflight Request and not reach invoke
-      if (!isInvokeRequest(body, ctx)) {
-        return new ctx.HTTPResponse({
-          statusCode: 400,
-          body: JSON.stringify({
-            error: 'Invoke request is not a JSON POST of the form { path, handler, body }',
-          }),
-          headers: { 'Content-Type': 'application/json' },
-        });
-      }
-      const { path, handler: fnHandler, args } = body;
-      let fn: Handler;
-      try {
-        fn = getHandler(backendDir, path, fnHandler);
-      } catch (error) {
-        if (error instanceof HandlerError) {
-          return new ctx.HTTPResponse({
-            statusCode: error.status,
-            body: JSON.stringify({ error: error.message }),
-            headers: { 'Content-Type': 'application/json' },
-          });
-        }
-        throw error;
-      }
-      const response = await fn(...args);
-      if (response === undefined) {
-        return new ctx.HTTPResponse({ statusCode: 204 });
+const app = express();
+app.post('/invoke', express.json(), async (req, res) => {
+  // TODO: check for allowed origins when supporting CORS
+  // Currently expected to 400 on a Preflight Request and not reach invoke
+  if (!isValidInvokeRequest(req)) {
+    return res.status(400).json({
+      error: 'Invoke request is not a JSON POST of the form { path, handler, body }',
+    });
+  }
+  const { path, handler: fnHandler, args } = req.body;
+  let fn: Handler;
+  try {
+    try {
+      fn = getHandler(backendDir, path, fnHandler);
+    } catch (error) {
+      if (error instanceof HandlerError) {
+        return res.status(error.status).json({ error: error.message });
       }
-      return response;
+      throw error;
     }
-    case 'sendStatus': {
-      return new ctx.HTTPResponse({
-        statusCode: decision.status,
-      });
-    }
-    case 'serveFile': {
-      const headers: Record<string, string> = {};
-      if (decision.contentType) {
-        headers['Content-Type'] = decision.contentType;
-      }
-      if (decision.cacheHint) {
-        Object.assign(headers, decision.cacheHint);
-      }
-      return new ctx.HTTPResponse({
-        statusCode: decision.status || 200,
-        headers,
-        body: await fs.readFile(decision.fullPath),
-      });
+    const response = await fn(...args);
+    if (response === undefined) {
+      return res.status(204).end();
     }
-    default: throw new Error('should not get here');
+    return res.status(200).json(response);
+  } catch (error) {
+    // tslint:disable-next-line:no-console
+    console.error('Failed to invoke handler', { error });
+    return res.status(500).json({ error: 'Failed to invoke' });
```

Comment:
We no longer send the error details to the client, which is probably a good default anyway.